### ISSUE TITLE
Propably wrong x,y coordinates on MouseMotionListener

### ISSUE THREAD

**Issue by f4n** [kwhat/jnativehook#267]
Hi,
i get some strange coordinates in the mouse move/drag event. I have a three monitor setup, two monitors have the same size, one is a little bit bigger. The screens are bottom aligned. It looks like this:

___ ___-----
| __| __| __|

The following code:
```
for( NativeMonitorInfo nm : GlobalScreen.getNativeMonitors() ) {
      System.out.println(nm.getNumber() + " " + nm.getWidth() + ":" + 
        nm.getHeight() + " " + nm.getX() + "x" + nm.getY());
}
```
gives me
```
0 1920:1080 0x120
1 1600:1200 3840x0
2 1920:1080 1920x120
```
Which means monitor 1 top left corner y offset is 120, monitor 2 y offset is 120  and monitor 3 y  offset is 0.
But now i see the mouse event coordinates on monitor 1 and 2 are in minimum 0 and on monitor 3 -120 (minus 120).

What i want to do is to detect on which monitor the mouse currently moves. Is there a bug or am i wrong that the y coordinate on monitor 1 and 2 should start at 120 and for monitor 3 by 0?

OS: Fedora 30
Library Version: 2.1.0

**Comment by kwhat** [comment]
iirc, the coordinates should all be relative to the primary displays upper left corner (0 x 0).  The primary display is always at monitor zero, but other displays like monitor 1 and 2 can be to the left or right of the primary display.

Each monitors coordinates should continue on relative to the primary display. For example: 

1. If you have two displays (both 1024 x 768), and the primary monitor is left of the secondary monitor, the coordinates for the top left corner of the secondary display should be 0 x 768.

2.  If you have the same setup, but the primary display is on the right of the secondary display,  the coordinates for the top left corner of the secondary display should be 0 x -768.

3. The same relative coordinates should apply also apply in the vertical direction.

Now, is this working correctly on your system?  I'm not exactly sure.  I will add this to my list of things to investigate further.

**Comment by kwhat** [comment]
Can you attach a screen shot of your monitor layout settings?  

Ex:
http://i.imgur.com/8PbRJVa.jpg
https://i.stack.imgur.com/wMcE9.png

I am curious why the primary display is offset by 120 px, there could be a settings "gap" between the displays.

**Comment by f4n** [comment]
You are correct. The coordinates rely to the primary display. As i changed the primary display to the screen with the biggest height, the y coordinates starting on this screen at 0, on the other screens at +120.

Thank you for your help. I will now write a offset detection in the application code to work around this. Maybe this is a feature which should be done in the library directly because i think this will help others too.

Original settings:
https://imgur.com/a/ATqifgn

**Comment by f4n** [comment]
And for other readers, my solution is to keep the offsets of the primary display (e.g. construcor/initializer).
```
int xOffset = 0;
int yOffset = 0;
if ( GlobalScreen.getNativeMonitors().length > 0 ) {
  NativeMonitorInfo primaryDisplay = GlobalScreen.getNativeMonitors()[0];
  xOffset = primaryDisplay.getX();
  yOffset = primaryDisplay.getY();
}
```
and in all callbacks of the library i apply the offset...
```
private void handeMovement(NativeMouseEvent e) {
    int x = e.getX() + xOffset;
    int y = e.getY() + yOffset;
    ....
}
```

**Comment by kwhat** [comment]
Alright I can confirm that this needs to be normalized on Linux.  Both the x,y coordinates as well as the monitor placement don't match other platforms.  We should fix the results produced by hook_create_screen_info and cache the result of the hook_create_screen_info on normalize the mouse coordinates.

This could be a breaking change so I am going to push it until 2.3
